src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs

```diff
@@ -125,7 +125,7 @@ internal void ClearCaches()
         {
             Debug.Assert(IsLockedInstance);
 
-            if (_cachingContext is null && _typeInfoResolver is not null)
```

Comment:
Second check no longer needed, since `IsLockedInstance` now implies that `typeInfoResolver` is not null,